qiskit_research/utils/pauli_twirling.py

```diff
@@ -43,31 +70,94 @@
     "secr": Pauli("XZ"),
 }
 
+
+def match_global_phase(a, b):
+    """Phase the given arrays so that their phases match at one entry.
+
+    Args:
+        a: A Numpy array.
+        b: Another Numpy array.
+
+    Returns:
+        A pair of arrays (a', b') that are equal if and only if a == b * exp(i phi)
+        for some real number phi.
+    """
+    if a.shape != b.shape:
+        return a, b
+    # use the largest entry of one of the matrices to maximize precision
+    index = max(np.ndindex(*a.shape), key=lambda i: abs(b[i]))
+    phase_a = cmath.phase(a[index])
+    phase_b = cmath.phase(b[index])
+    return a * cmath.rect(1, -phase_a), b * cmath.rect(1, -phase_b)
+
+
+def allclose_up_to_global_phase(op1, op2, tol=1e-15):
+    """Check if two operators are close up to a global phase."""
+    # Phase both operators to match their phases
+    op1_array = op1.to_matrix()
+    op2_array = op2.to_matrix()
+
+    phased_op1, phased_op2 = match_global_phase(op1_array, op2_array)
+    return np.allclose(phased_op1, phased_op2, atol=tol)
+
+
+def create_pauli_twirling_sets(two_qubit_gate):
+    """Generate the Pauli twirling sets for a given 2Q gate.
+
+    Sets are ordered such that gate[0] and gate[1] are pre-rotations
+    applied to control and target, respectively. gate[2] and gate[3]
+    are post-rotations for control and target, respectively.
+
+    Parameters:
+        two_qubit_gate (Gate): Input two-qubit gate
+
+    Returns:
+        tuple: Tuple of all twirling gate sets
+    """
+
+    # Generate 16-element list of Pauli gates, each repeated 4 times
+    target_unitary = Operator(two_qubit_gate.to_matrix())
```

Comment:
```suggestion
    target_unitary = np.array(two_qubit_gate)
```
I think this works (?)